### ISSUE TITLE
[Request] Handle mail without preview images

### ISSUE THREAD

**Issue by ronaldheft** [moralmunky/Home-Assistant-Mail-And-Packages#114]
I've noticed in my inbox today, I received an email with two preview images, and a third preview image that looks like the below:

![IMG_1877](https://user-images.githubusercontent.com/1143112/81591701-d82fc280-938a-11ea-9923-07f9b5a7b998.png)

I understand that Informed Delivery doesn't always include every letter (particularly when dealing with junk mail), but I think it would be awesome keep Home Assistant's sensors as close to the email as possible. So, if USPS reports on something that can't be scanned, we include in the count.

My proposal:

- When USPS includes in the email that there was a letter than couldn't be scanned, include it the count.

Undecided:

- If the preview image that USPS provides should be included in the GIF. My vote would be yes, as when there is only 1 piece of mail that hits this scenario, it may be weird to see 1 mail on the sensor and then a GIF with "No Mail Today".

For parsing, here is the RAW email contents of that section. Happy to provided more of the email privately for development purposes:

```
<table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#F7F=
7F7" style=3D"margin-top: 10px;margin-bottom:20px;padding-top: 10px;padding=
-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:30px;padding-left:50px; padding-right:50px;width:800px !import=
ant;margin-bottom:20px;">=20
    <tr style=3D"text-align:center">
     <td align=3D"center" width=3D"700" colspan=3D"2" style=3D"padding-top:=
30px; margin: 0 auto;margin-bottom:20px;display:inline-block;text-align:cen=
ter;width:700px !important;"><img width=3D"700" style=3D"vertical-align: mi=
ddle;margin: 0 auto;border:none;width:700px !important;text-align:center;" =
src=3D"https://www.usps.com/email/id/image-no-mailpiece700.jpg" alt=3D"A ma=
ilpiece for which we do not currently have an image is included in today=EF=
=BF=BDs mail." /></td>
=20
  </tr>
 =20
  </table><!--end table content-->
  </td></tr>
   =20
</table>=20
```

**Comment by firstof9** [comment]
> * When USPS includes in the email that there was a letter than couldn't be scanned, include it the count.

It does get counted.

The image is the only thing that is left out. I'm sure we could turn that off and insert the "Mail Image Missing" image.

@moralmunky what do you think? We could even make it a toggle in the options. ie: "Include Missing Mail Image?" as a bool.

**Comment by moralmunky** [comment]
This section of the code is looking for that missing image graphic and should be including it in the generated gif.

https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/d9f7fe161ea3ed9f2148c1a4ea165faa2d69a734/custom_components/mail_and_packages/sensor.py#L433

**Comment by moralmunky** [comment]
I doubled checked this with a recent email that had 3 images and a non image. It counted 4 and stitched the no image graphic into the GIF. 

Certain HASS components that display images reload the file every 10 seconds or so, if the 'Time in seconds per image' time the number of images that needs to display is greater than 10 you won't see all the images the gif. This is not an issue if you use the custom card to display the mail information.

**Comment by ronaldheft** [comment]
Interesting. The email I received today has two images and a non image. My sensor counted 2 and the GIF only shows the 2 images (I’m viewing the GIF directly to confirm).

One of the mail pieces received today was from the IRS, and there was a special IRS section in the email. Maybe that section messed things up? The IRS letter did show up in the GIF.

Here is the full HTML (realized it’s okay to include as the images are attachments and I removed some links):

```
------=_Part_24594_1790679384.1589196824414
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: quoted-printable

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.=
w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns=3D"http://www.w3.org/1999/xhtml"
 xmlns:v=3D"urn:schemas-microsoft-com:vml"
 xmlns:o=3D"urn:schemas-microsoft-com:office:office" lang=3D"en">
<head>
<meta http-equiv=3D"Content-Type" content=3D"text/html; charset=3DUTF-8" />
<meta name=3D=E2=80=9Dx-apple-disable-message-reformatting=E2=80=9D>
<meta name=3D"viewport" content=3D"width=3Ddevice-width, initial-scale=3D1.=
0"/>
<title>Informed Delivery</title>

<!--[if gte mso 9]><xml>
 <o:OfficeDocumentSettings>
  <o:AllowPNG/>
  <o:PixelsPerInch>96</o:PixelsPerInch>
 </o:OfficeDocumentSettings>
</xml><![endif]-->

<style type=3D"text/css">
a{
color:#333366;text-decoration:none;font-weight: bold;
}
a:hover{
color:#999999;text-decoration:none;font-weight: bold;
}
a:focus{
color:#999999;text-decoration:none;font-weight: bold;
}
td a{
color:#333366;text-decoration:none;font-weight: bold;
}
td a:hover{
color:#999999;text-decoration:none;font-weight: bold;
}
td a:focus{
color:#999999;text-decoration:none;font-weight: bold;
}=09
p {
=09margin: 0;
=09padding: 0;
}
=09
table {
=09margin:0;
=09padding:0;
=09border-collapse: collapse;
}
=20
.cleardiv {
    clear:both;
}

.redback {
background-color: #e71a21; /* Old browsers */
bgcolor: #e71a21; /* Old browsers */
}

.amberback {
background-color: #f9cc00; /* Old browsers */
bgcolor: #f9cc00; /* Old browsers */
}
   =20
.greenback {
background-color: #73f55f !important; /* Old browsers */
bgcolor: #73f55f !important; /* Old browsers */
}

.blueback {
background-color: #0e84f9 ; /* Old browsers */
bgcolor: #0e84f9; /* Old browsers */
}

.purpleback {
background-color: #c511fc; /* Old browsers */
bgcolor: #c511fc; /* Old browsers */
}
=09


</style>

 <!--[if mso]>
  <style type=3D"text/css">  =20
.redback {
background-color: #ffd3d3; /* Old browsers */
}

.amberback {
background-color: #f9cc00 !important;=20
bgcolor: #f9cc00 !important;=20
}
   =20
.greenback {
background-color: #73f55f !important;
bgcolor: #73f55f !important;=20
}

.blueback {
background-color: #0e84f9 !important;=20
bgcolor: #0e84f9;=20
}

.purpleback {
background-color: #c511fc !important;=20
bgcolor: #c511fc !important;
}
  </style>
<![endif]-->
</head>

<body style=3D"margin: 0 auto; font-family:Arial, sans-serif; font-size: 16=
px;color:#595959; width: 100%;padding:0;" lang=3D"EN-US" link=3D"#333366" v=
link=3D"#333366">

<!--main table-->
<table role=3D"presentation" align=3D"center" width=3D"100%" border=3D"0" c=
ellspacing=3D"0" cellpadding=3D"0" style=3D"padding:10px;margin:10px;margin=
:0 auto;width:100% !important;" >
<tr>
<td>
<table role=3D"presentation" align=3D"center" width=3D"800" border=3D"0" ce=
llspacing=3D"0" cellpadding=3D"0" style=3D"border-spacing: 0px;-webkit-bord=
er-horizontal-spacing: 0px; -webkit-border-vertical-spacing: 0px; font-fami=
ly:Arial, sans-serif; font-family:Arial, sans-serif; font-size: 0.750em;col=
or:#000000; width:800;margin:0 auto;">


<tr><!--row within main table-->
<td width=3D"800">
=09<table role=3D"presentation" width=3D"800" cellpadding=3D"0" cellspacing=
=3D"0" style=3D"border-collapse: collapse; font-family:Arial, sans-serif;wi=
dth:800px">
=09=09<tr>
=09=09<td width=3D"50%" align=3D"center" style=3D"padding-top: 20px; paddin=
g-bottom: 30px; font-size: 12px; font-family: Arial, sans-serif; padding-ri=
ght: 5px; color: #595959; line-height: 16px; text-align: left;">You have ma=
il and packages arriving soon.</td>
=09=09<td width=3D"50%" align=3D"center" style=3D"padding-top: 20px; paddin=
g-bottom: 30px; font-size: 12px; font-family: Arial, sans-serif; padding-ri=
ght: 5px; color: #595959; line-height: 16px; text-align: right;"><span styl=
e=3D"padding-top: 20px; padding-bottom: 30px; font-size: 12px; font-family:=
 Arial, sans-serif; padding-right: 5px; color: #595959; line-height: 16px; =
text-align: left;">5/11/2020</span></td>
=09=09</tr>
=09</table>
=09
=09<table role=3D"presentation" align=3D"center" width=3D"800" cellpadding=
=3D"0" cellspacing=3D"0">
=09=09<tr>
=09=09=09<td align=3D"left" valign=3D"middle" style=3D"vertical-align:middl=
e;padding-left:30px"><span><a href=3D"https://www.usps.com/" style=3D"borde=
r:none; font-family:Arial, sans-serif;vertical-align:middle" tabindex=3D"1"=
><img src=3D"https://www.usps.com/real-mail/images/logo_mobile.png" width=
=3D"64" style=3D"border:none;" alt=3D"USPS.com Informed Delivery" id=3D"log=
o--main"/></a></span>
=09=09=09<span style=3D"text-align:center;font-family:Arial, sans-serif; fo=
nt-size: 36px; color:#333366; line-height: 34px; font-weight: bolder;vertic=
al-align:top;padding-left:20px">COMING TO YOUR MAILBOX SOON.</span></td>
=09    </tr>
=09  =20
=09 =20
=09
=09 </table>
 =20
 <!--  customerAlerts BEGIN -->=20
        <div align=3D"center">      =20
            =20
        </div>    =20
<!--  customerAlerts END -->

 <br />
   <div width=3D"800" align=3D"center" style=3D"margin-top: 30px;">
   <img src=3D"https://www.usps.com/email/id/IDdailyDmailHeader.jpg" alt=3D=
"You have mail arriving soon. " width=3D"800" height=3D"116" style=3D"borde=
r:none;" title=3D"You have mail arriving soon. "/>
   </div><br />
      <table role=3D"presentation" align=3D"center" width=3D"800" >
   =09 <tr>=20
=09=09=09<td style=3D"text-align:right;vertical-align:middle"><a href=3D"ht=
tps://informeddelivery.usps.com/box/pages/secure/DashboardAction_input?keyw=
ord=3Dmail" style=3D"vertical-align: middle; font-family:Arial, sans-serif;=
color:#333366;font-weight:bold;font-size:16px;text-decoration:none; line-he=
ight: 18px;" tabindex=3D"2">View all mail on dashboard&nbsp;<img src=3D"htt=
ps://www.usps.com/real-mail/images/red-caret-low-res.png" width=3D"10" heig=
ht=3D"12" style=3D"vertical-align: middle; padding-left: 1px; margin-top: -=
1px;border:none;" alt=3D"red arrow"></a></td>
=09=09</tr>=20
   </table>
   <!--MAILPIECES AND RIDEALONG HTML FIT IN HERE BEGIN-->=20
 <div align=3D"center" style=3D"margin-top: 30px">
  =20
<table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#F7F=
7F7" style=3D"margin-top: 10px;margin-bottom:20px;padding-top: 10px;padding=
-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:30px;padding-left:50px; padding-right:50px;width:800px !import=
ant">=20
    <tr style=3D"margin-bottom:20px;text-align:center">
     <td align=3D"center" width=3D"700" colspan=3D"2" style=3D"padding-bott=
om:20px;padding-top:30px; border:none;vertical-align: middle;margin: 0 auto=
;margin-bottom:20px;text-align:center;width:700px !important;"><img width=
=3D"700" style=3D"width: 700px !important;vertical-align: middle;margin: 0 =
auto;border:none;text-align:center;" src=3D"cid:1102783650-018.jpg_67174409=
23315076" alt=3D"Scanned image of your mail piece" /></td>
  </tr>
   <tr width=3D"800" cellpadding=3D"0" cellspacing=3D"0"
=09style=3D"width: 800px; background-color: #f7f7f7; border: none; text-ali=
gn: left; margin: 0 auto; margin-top: 10px; margin-bottom: 10px; font-famil=
y: Arial, sans-serif; font-size: 1em; color: #000000;"
=09align=3D"center">



=09<td width=3D"270"
=09=09style=3D"padding-right: 20px; border: none; padding-top: 0px; padding=
-bottom: 0px;padding-left:50px;padding-bottom:30px"
=09=09align=3D"left" cellpadding=3D"0" cellspacing=3D"0">
=09=09
=09=09 <a href=3D""><img + width=3D'250' + src=3D"c=
id:ra_0_6717440923595914"></a>
=09=09
=09=09</td>
=09<td width=3D"530" style=3D"width: 530px !important; padding-top: 15px; p=
adding-bottom: 30px;"
=09=09align=3D"left"><font face=3D"Arial, sans-serif"
=09=09cellpadding=3D"0" cellspacing=3D"0">
=09=09=09<div style=3D"margin-top: 15px; font-size: 1em;color: #595959;">
=09=09=09=09FROM <span style=3D"color: #595959; font-weight: bold; margin: =
10px 0px; text-decoration: none; margin-left: 15px; margin-bottom: 5px; fon=
t-size: 1em;">IRS</span>
=09=09=09</div> <br>
=09=09<br>
=09=09=09<div style=3D"color: #333366; font-weight: bold; text-decoration: =
none; font-size: 1em; display: block; margin-top: 5px; vertical-align: midd=
le;" alt=3D"image of ride along content for your mail piece">
=09=09=09=09<span><a href=3D"">Learn More</a></span>&nbsp;<a href=
=3D""
=09=09=09=09=09style=3D"text-decoration: none; color: #333366; border: none=
; vertical-align: middle;"><img
=09=09=09=09=09src=3D"https://www.usps.com/real-mail/images/red-caret-low-r=
es.png"
=09=09=09=09=09width=3D"10" height=3D"12" alt=3D"link to ride along website=
"
=09=09=09=09=09style=3D"border: none; padding-left: 12px; padding-top: 2px;=
 margin-bottom: -2px; vertical-align: middle; background-color: transparent=
" /></a>
=09=09=09</div>
=09</font></td>
</tr>


  </table><!--end table content-->
  </td></tr>
   =20
</table>

<table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#F7F=
7F7" style=3D"margin-top: 10px;margin-bottom:20px;padding-top: 10px;padding=
-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:30px;padding-left:50px; padding-right:50px;width:800px !import=
ant">=20
    <tr style=3D"margin-bottom:20px;text-align:center">
     <td align=3D"center" width=3D"700" colspan=3D"2" style=3D"padding-bott=
om:20px;padding-top:30px; border:none;vertical-align: middle;margin: 0 auto=
;margin-bottom:20px;text-align:center;width:700px !important;"><img width=
=3D"700" style=3D"width: 700px !important;vertical-align: middle;margin: 0 =
auto;border:none;text-align:center;" src=3D"cid:1102783452-018.jpg_67174409=
34567516" alt=3D"Scanned image of your mail piece" /></td>
  </tr>
 =20
  </table><!--end table content-->
  </td></tr>
   =20
</table>
<table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#F7F=
7F7" style=3D"margin-top: 10px;margin-bottom:20px;padding-top: 10px;padding=
-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:30px;padding-left:50px; padding-right:50px;width:800px !import=
ant;margin-bottom:20px;">=20
    <tr style=3D"text-align:center">
     <td align=3D"center" width=3D"700" colspan=3D"2" style=3D"padding-top:=
30px; margin: 0 auto;margin-bottom:20px;display:inline-block;text-align:cen=
ter;width:700px !important;"><img width=3D"700" style=3D"vertical-align: mi=
ddle;margin: 0 auto;border:none;width:700px !important;text-align:center;" =
src=3D"https://www.usps.com/email/id/image-no-mailpiece700.jpg" alt=3D"A ma=
ilpiece for which we do not currently have an image is included in today=EF=
=BF=BDs mail." /></td>
=20
  </tr>
 =20
  </table><!--end table content-->
  </td></tr>
   =20
</table>=20
   </div><br />

<!--MAILPIECES AND RIDEALONG HTML FIT IN HERE END-->
  =20
   <!--PACKAGES HTML FIT IN HERE BEGIN-->=20
  <br />=20
   <div  align=3D"center" style=3D"margin-top: 30px">
   <img src=3D"https://www.usps.com/email/id/IDdailyDpackagesHeader.jpg" al=
t=3D"The packages below are expected to be delivered based upon data at the=
 time this email was sent." width=3D"800" height=3D"116" style=3D"border:no=
ne;" title=3D"The packages below are expected to be delivered based upon da=
ta at the time this email was sent."/>
   </div><br />
  =20
  =09<table role=3D"presentation" align=3D"right" >
   =09 <tr>=20
=09=09=09<td style=3D"text-align:right;vertical-align:middle;width:100%;pad=
ding-bottom:25px;"><a href=3D"https://informeddelivery.usps.com/box/pages/s=
ecure/DashboardAction_input?keyword=3Dpack" style=3D"text-align:right;verti=
cal-align: middle; font-family:Arial, sans-serif;color:#333366;font-weight:=
bold;font-size:16px;text-decoration:none; line-height: 18px;" tabindex=3D"2=
">View all packages on dashboard&nbsp;<img src=3D"https://www.usps.com/real=
-mail/images/red-caret-low-res.png" width=3D"10" height=3D"12" style=3D"ver=
tical-align: middle; padding-left: 1px; margin-top: -1px;border:none;" alt=
=3D"red arrow"></a></td>
=09=09</tr>=20
   </table>
   =20
     =20
      <br />
   <div align=3D"left" width=3D"100%" style=3D"margin-top: 20px;padding-lef=
t:20px">
      <br />=20
  =20
  =20
  =09<table role=3D"presentation" align=3D"left" width=3D"100%" style=3D"ma=
rgin-bottom:30px">
   =09 <tr>=20
=09=09=09<td align=3D"left" style=3D"text-align:left;vertical-align:middle;=
padding-left:20px"><span style=3D"vertical-align: middle; font-family:Arial=
, sans-serif;color:#333366;font-weight:bold;font-size:20px;text-decoration:=
none; line-height: 18px;padding-right:0px;margin-right:0px">Arriving Today<=
/span></td>
=09=09=09<td align=3D"right" style=3D"text-align:right;vertical-align:middl=
e;padding-right:20px"><span style=3D"vertical-align: middle; font-family:Ar=
ial, sans-serif;color:#595959;font-weight:normal;font-size:20px;text-decora=
tion:none; line-height: 18px;padding-right:0px;margin-right:0px;">Monday, M=
ay 11</span></td>
=09=09</tr>=20
=09=09<tr>
=09=09  <td valign=3D"middle" align=3D"center" colspan=3D"2" style=3D"font-=
family:Arial, sans-serif;font-size: 11px;padding-left:0px;padding-top:10px"=
><hr size=3D"2" style=3D"margin:0 auto; padding-left:0px; font-family:Arial=
, sans-serif;font-size: 12px;color:#333366;"></td>
=09    </tr>
   </table>
   =20
     =20
   <div align=3D"left" style=3D"margin-top: 20px;">
   <table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#=
ffffff" style=3D"margin-top: 10px;margin-bottom:0px;padding-top: 10px;paddi=
ng-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:20px;width:800px !important">=20
    <tr style=3D"margin-bottom:20px;">
     <td align=3D"left" style=3D"padding-top:30px; padding-bottom:0px; font=
-size: 16px; text-align: left; font-family:Arial, sans-serif;vertical-align=
:middle;color:#595959; line-height: 18px;padding-left:40px">No packages ava=
ilable to display.</td>
  </tr>
  </table><!--end table content-->
  </td></tr>
   =20
</table>  =20
=20
   </div><br />
  =20
   =20
   </div><br />
  =20
   <div align=3D"left" style=3D"margin-top: 40px;padding-left:20px">
    <br />=20
  =20
  =20
  =09<table role=3D"presentation" align=3D"left" width=3D"100%" style=3D"ma=
rgin-bottom:30px">
   =09 <tr>=20
=09=09=09<td align=3D"left" style=3D"text-align:left;vertical-align:middle;=
padding-left:20px"><span style=3D"vertical-align: middle; font-family:Arial=
, sans-serif;color:#333366;font-weight:bold;font-size:20px;text-decoration:=
none; line-height: 18px;padding-right:0px;margin-right:0px">Arriving Soon</=
span></td>
=09=09</tr>=20
=09=09<tr>
=09=09  <td valign=3D"middle" align=3D"center" colspan=3D"1" style=3D"font-=
family:Arial, sans-serif;font-size: 11px;padding-left:0px;padding-top:10px"=
><hr size=3D"2" style=3D"margin:0 auto; padding-left:0px; font-family:Arial=
, sans-serif;font-size: 12px;color:#333366;"></td>
=09    </tr>
   </table>
   =20
  =20
  <div align=3D"left" style=3D"margin-top: 20px;">
   <table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#=
ffffff" style=3D"margin-top: 10px;margin-bottom:0px;padding-top: 10px;paddi=
ng-bottom:20px;width:800px !important">
  <tr>
    <td>
    <table role=3D"presentation" width=3D"800" align=3D"center" style=3D"pa=
dding-bottom:20px;width:800px !important">=20
    <tr style=3D"margin-bottom:20px;">
     <td align=3D"left" style=3D"padding-top:30px; padding-bottom:0px; font=
-size: 16px; text-align: left; font-family:Arial, sans-serif;vertical-align=
:middle;color:#595959; line-height: 18px;padding-left:40px">No packages ava=
ilable to display.</td>
  </tr>
  </table><!--end table content-->
  </td></tr>
   =20
</table>  =20
=20
   </div><br />
  =20
  =20
 =20
 =20
   </div><br />
  =20
=20
<!--PACKAGES HTML FIT IN HERE END-->
   =20
   =20
    =20
   <!--table Go to Dashboard link BEGIN-->
    =20
=09<table role=3D"presentation" align=3D"center" width=3D"800">=20
=09<tr>
=09<td align=3D"center" width=3D"800" style=3D"padding-top:15px; padding-bo=
ttom:15px;"></td></tr>
</table>=20
  =20
   <table role=3D"presentation" align=3D"center" width=3D"800" bgcolor=3D"#=
F7F7F7" style=3D"font-family:Arial, sans-serif;background-color:#f7f7f7;bor=
der-width: 1px;border-color: #595959;border-style:solid;">=20
=09
=09<tr>
=09<td align=3D"center" width=3D"800" style=3D"padding-top:30px; padding-bo=
ttom:30px; font-size: 16px; text-align: center; font-family:Arial, sans-ser=
if;vertical-align:middle;color:#595959; line-height: 18px;">
=09=09You may have more mail or packages than are shown in your Daily Diges=
t. To check, <a href=3D"https://informeddelivery.usps.com/box/pages/secure/=
DashboardAction_input?keyword=3Dmail" tabindex=3D"3" style=3D"color:#333366=
;font-weight:bold;font-size:16px;text-decoration:none;">go to your Dashboar=
d&nbsp;&nbsp;<img src=3D"https://www.usps.com/real-mail/images/red-caret-lo=
w-res.png" width=3D"10" height=3D"12" style=3D"vertical-align: middle; padd=
ing-left: 2px; margin-top: -4px; font-family:Arial, sans-serif;border:none;=
" alt=3D"red arrow"></a>

=09</td></tr>

=09</table>=20
    <!--table Go to Dashboard link END-->
  =20
              =09
   <!--table Didnt get a mailpiece-->
=09<table role=3D"presentation" width=3D"800" align=3D"center" style=3D"mar=
gin-top: 0px; font-family:Arial, sans-serif;">=20
  =20
    <!--row within container table-->
=09<tr><td width=3D"800" align=3D"center" style=3D"padding-top:30px; paddin=
g-bottom:30px; font-size:16px; text-align: center; font-family:Arial, sans-=
serif;vertical-align:middle;color:#595959; line-height: 20px;">
=09Mail may arrive several days after you receive this notification.  Pleas=
e allow up to a week for delivery before reporting missing mail.<br><br>
=09<div>
=09<a href=3D"" tabindex=
=3D"4" style=3D"color:#333366;font-weight:bold;font-size:16px;text-decorati=
on:none;">Report missing mail&nbsp;&nbsp;<img src=3D"https://www.usps.com/r=
eal-mail/images/red-caret-low-res.png" width=3D"10" height=3D"12" style=3D"=
vertical-align: middle; padding-left: 2px; margin-top: -4px; font-family:Ar=
ial, sans-serif;border:none;" alt=3D"red arrow"></a></div>

=09</td></tr>
=09
=09 <tr>
=09=09  <td valign=3D"middle" align=3D"center" colspan=3D"2" style=3D"paddi=
ng-bottom:25px; font-family:Arial, sans-serif;font-size: 11px;"><hr size=3D=
"2" style=3D"margin:0 auto; padding-left:20px; font-family:Arial, sans-seri=
f;font-size: 12px;color:#333366;"></td>
=09    </tr>
=09   =20
=09=09<tr>
=09=09<td style=3D"font-size: 11px; font-family:Arial, sans-serif; color:59=
5959; line-height: 16px; padding-bottom:30px;padding-left:20px;padding-righ=
t:20px;" ><p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Aria=
l, sans-serif; color:595959; line-height: 14px;">*Not all mail is imaged an=
d sorted on USPS<sup>&reg;</sup> automated equipment. Therefore, some of yo=
ur mail may not be shown here. </p>
=09=09  <p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial,=
 sans-serif; color:595959; line-height: 14px;">You subscribed to this servi=
ce with USPS<sup>&reg;</sup> New Products &amp; Innovation, PO Box 23972, W=
ashington DC 20026-3972.</p>
=09=09=09
=09=09=09<p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial=
, sans-serif; color:595959; line-height: 14px;">If you no longer wish to re=
ceive daily email notifications, <a href=3D"" styl=
e=3D"font-size: 11px; font-weight: bold; color: #333366; font-family:Arial,=
 sans-serif;text-decoration:none;" tabindex=3D"5">unsubscribe here.</a> </p=
>
=09=09=09
=09=09=09<p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial=
, sans-serif; color:595959; line-height: 14px;">If you need support, please=
 visit <a href=3D"https://uspshelp.custhelp.com/app/ask_id" style=3D"font-s=
ize: 11px; font-weight: bold; color: #333366; font-family:Arial, sans-serif=
;text-decoration:none;" tabindex=3D"6">user support for Informed Delivery<s=
up>&reg;</sup>.</a> </p>
=09=09=09
=09=09=09<p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial=
, sans-serif; color:595959; line-height: 14px;">For more information about =
this service, please visit <a href=3D"http://faq.usps.com" style=3D"font-si=
ze: 11px; font-weight: bold; color: #333366; font-family:Arial, sans-serif;=
text-decoration:none;" tabindex=3D"7">general information about Informed De=
livery.</a> </p>
=09=09=09<p style=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial=
, sans-serif; color:595959; line-height: 14px;">Copyright&copy; 2018 United=
 States Postal Service<sup>&reg;</sup>. All Rights Reserved. The Eagle Logo=
 and the trade dress of USPS<sup>&reg;</sup> Packaging are among the many t=
rademarks of the U.S. Postal Service<sup>&reg;</sup>. </p>=09=09=09<p style=
=3D"margin-bottom: 15px;font-size: 11px; font-family:Arial, sans-serif; col=
or:595959; line-height: 14px;">This is an automated email, please do not re=
ply to this message. This message is for the designated recipient only and =
may contain privileged, proprietary, or otherwise private information. If y=
ou have received it in error, please delete. Any other use of the email by =
you is prohibited.</p>
=09=09=09
=09=09=09</td>
=09=09</tr>
=09=09
=09=09

=09</table> <!--end table-->
  =20


</td></tr><!--end row within main table-->
=20
</table>
</td>
</tr>
</table><!--end main table-->
<p style=3D"display:none;visibility:hidden">
<br />
<img height=3D"0" width=3D"0" alt=3D"" src=3D"" styl=
e=3D"display:none;visibility:hidden" />
<img height=3D"0" width=3D"0" alt=3D"" src=3D"" style=3D"display:none;visi=
bility:hidden" />
</p>
</body>
</html>

------=_Part_24594_1790679384.1589196824414
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

Informed Delivery(TM)

Today's Deliveries: 5/11/2020

You can view your mailpieces online at the following URL:
https://informeddelivery.usps.com/box/pages/secure/DashboardAction_input?keyword=mail

You subscribed to this service with USPS(R) New Products & Innovation, PO Box 23972, Washington DC 200263972.

You can unsubscribe at the following URL:

Copyright(C) 2018 United States Postal Service(R). All Rights Reserved. Informed Delivery(TM) and the trade dress of USPS(R) Packaging are among the many trademarks of the U.S. Postal Service(R).

This is an automated message. Please do not reply.

```

**Comment by ronaldheft** [comment]
@firstof9 @moralmunky Actually I see this issue! Looks like the image was renamed. The code is searching for:

```
image-no-mailpieces700.jpg
```

And the email I received has the image as (not plural):

```
image-no-mailpiece700.jpg
```

Maybe check for both types, in the event they’re A/B testing something?

**Comment by moralmunky** [comment]
Great catch! The email I tested with was from May 9th. I’ll have to recheck what the content of the image was for my test to see if the message in the image was different.

It’s possible they have had two versions of the image all along.

**Comment by firstof9** [comment]
Shouldn't be too hard to account for both. I'll look at it this afternoon.

**Comment by firstof9** [comment]
@ronaldheft Give this file a shot: https://raw.githubusercontent.com/firstof9/Home-Assistant-Mail-And-Packages/condense-sensor/custom_components/mail_and_packages/sensor.py

**Comment by ronaldheft** [comment]
@firstof9 That regex works! I just copied over the regex change, as there were some other changes in your `sensor.py` file (running off the latest `master`).

Both the sensor and the GIF now include the non-image mail piece.

I should say, however, it looks like I did hit issue #32 in the GIF (the two letters were the correct size, the non-image was cropped off), but that’s a separate issue.

![52AE3BC3-C442-40DB-A146-A3263114D921](https://user-images.githubusercontent.com/1143112/81622591-5dcd6580-93bf-11ea-938a-01894d7def2d.jpeg)
